spec/03-plugins/06-statsd/01-log_spec.lua

```diff
@@ -655,6 +665,37 @@ for _, strategy in helpers.each_strategy() do
     end)
 
     describe("metrics", function()
+      -- the purpose of this test case is to test the batch queue
+      -- finishing processing its batch in one time (no retries)
+      it("no more metrics than expected", function()
+        local metrics_count = DEFAULT_METRICS_COUNT + shdict_count * 2
+
+        local thread = helpers.udp_server(UDP_PORT, metrics_count + 1, 5)
+        local response = assert(proxy_client:send {
+          method  = "GET",
+          path    = "/request?apikey=kong",
+          headers = {
+            host  = "logging31.com"
+          }
+        })
+        assert.res_status(200, response)
+
+        local ok, metrics, err = thread:join()
+        assert(ok, metrics)
+        assert(#metrics == metrics_count, err)
+      end)
+
+      -- wait until shdict metrics could be sent again
```

Comment:
Will this introduce flakiness? I am good if it isn't. 🤔

Reply:
This test case aims to ensure the changes(add return values) for functions of statsd plugins called in batch_queue won't be called repeatedly when the functions executed successfully. In this test case, I try to wait more metrics than it should be, actually , it should timeout as expected (because no more metrics then expected). 
The next one is waiting for the shdict metrics being refreshed so that the third test case can pass.
I tried several times in my local environment successfully, so I think it won't be flaky.